agent/api/appmesh/appmesh.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package appmesh
+
+import (
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/acs/model/ecsacs"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+const (
+	appMesh            = "APPMESH"
+	splitter           = ","
+	ignoredUID         = "IgnoredUID"
+	ignoredGID         = "IgnoredGID"
+	proxyIngressPort   = "ProxyIngressPort"
+	proxyEgressPort    = "ProxyEgressPort"
+	appPorts           = "AppPorts"
+	egressIgnoredIPs   = "EgressIgnoredIPs"
+	egressIgnoredPorts = "EgressIgnoredPorts"
+)
+
+// AppMesh contains information of app mesh config
+type AppMesh struct {
+	// App Mesh CNI enabled
+	AppMeshEnabled bool
```

Comment:
why exactly do we need this field? 

couldn't we find out if a task is appmesh enabled if AppMesh != nil?

Reply:
make sense, i can check nil to determine app mesh is enabled